cli.js

```diff
@@ -9,46 +9,77 @@ const express = require('express')
 const graphqlHTTP = require('express-graphql')
 const OasGraph = require('./lib/index.js')
 const path = require('path')
+const http = require('http')
+const fs = require('fs');
 
-const app = express()
-const [,, ... args] = process.argv
-
-// Check for arguments
-if (args.length > 0) {
-  let filePath = args[0]
-
-  // Convert path to absolute path
-  // Otherwise, would need to have separate cases for relative or absolute paths
-  filePath = path.resolve(filePath)
-
-  try {
-    let oas = require(filePath)
-
-    // Create GraphQL interface
-    OasGraph.createGraphQlSchema(oas)
-      .then(({schema, report}) => {
-        console.log(JSON.stringify(report, null, 2))
-        app.use('/graphql', graphqlHTTP({
-          schema: schema,
-          graphiql: true
-        }))
-
-        app.listen(3001, () => {
-          console.log('GraphQL accessible at: http://localhost:3001/graphql')
-        })
-      })
-      .catch(err => {
-        console.log(err)
-      })
-  } catch (error) {
-    if (error.code === 'MODULE_NOT_FOUND') {
-      console.log(`The file path "${filePath}" is invalid.`)
+const app = express();
+
+if (process.argv.length <= 2) {
+     console.log("Usage: " + __filename + " <OAS JSON file path or remote url> [port number]");
+     process.exit(-1);
+}
+
+let filePath = process.argv[2];
+let portNumber = process.argv[3] ? process.argv[3] : 3001;
+
+// check if the file exists 
+if (fs.existsSync(path.resolve(filePath))) {
+      let oas = JSON.stringify(require(path.resolve(filePath)));
+      startGraphQLServer(oas, portNumber);
+    
+} else { // falls back to a remote location
+    if (filePath.includes('http')) {
+        getRemoteFileSpec(filePath).then(remoteContent=> {
+          startGraphQLServer(remoteContent, portNumber);
+        });
     } else {
-      console.log(error)
-    }
-  }
-} else {
-  console.log(
-    'Usage: oasgraph <OAS JSON file path>'
-  )
+        console.log(`OASGraph reading local file error. file ${filePath} does not exist.`);
+      } 
+}
+
+/**
+ * reads a remote file content using http protocol
+ * @param {string} url specifies a valid URL path including the port number
+ */
+function getRemoteFileSpec (url) { 
+  return new Promise(resolve => { 
+    http.get(url, (resp) => {
```

Comment:
The built-in `http` module does not support https. 

Because we already use the `request` module, @ErikWittern proposed we replace the code with the following:

```
function getRemoteFileSpec (uri) {
  return new Promise((resolve, reject) => {
    request({
      uri,
      json: true
    }, (err, res, body) => {
      if (err) {
        reject(err)
      } else if (res.statusCode !== 200) {
        reject(new Error(`Error: ${JSON.stringify(body)}`))
      } else {
        resolve(body)
      }
    })
  })
}
```